.github/workflows/go-scheduled.yml

```diff
@@ -31,17 +39,50 @@ jobs:
             dep ensure
         fi
 
+    - name: Format
+      run: |
+        diff -u <(echo -n) <(gofmt -d .)
+
+    - name: Lint
+      run: |
+        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.30.0
+        /home/runner/go/bin/golangci-lint run --skip-dirs=pkg/promql --skip-dirs=pkg/promb
+
     - name: Build
       run: go build -v ./...
 
-    - name: Test
-      run: go test -v -race ./...
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
 
-    - name: Test W/O Promscale Extension
-      run: go test -v -race ./pkg/tests/end_to_end_tests/ -use-extension=false
+    - name: Set up Go 1.14
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
 
-    - name: Test W/O TimescaleDB Extension
-      run: go test -v -race ./pkg/tests/end_to_end_tests/ -use-extension=false -use-timescaledb=false
+    - name: Use Go module caching
+      uses: actions/cache@v2
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
```

Comment:
since we don't use dep, can we remove this?

Reply:
I guess we can, I didn't want to touch it for this PR.